In [2]:
# Load the required packages
import numpy as np
import scipy as sc
import matplotlib as plt
import seaborn as sns
import pandas as pd
import statsmodels as sm
import sklearn as sk

## Resources

Task 1
https://scikit-learn.org/1.4/auto_examples/datasets/plot_iris_dataset.html

Task 2
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.shape.html,  
https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_iris.html,  


Task 3
https://numpy.org/doc/2.2/reference/arrays.ndarray.html
https://www.datacamp.com/tutorial/python-dictionary-append
https://numpy.org/doc/2.1/reference/generated/numpy.median.html
https://docs.python.org/3/library/functions.html
https://docs.python.org/3/tutorial/inputoutput.html



# Task 1: Source the Data Set
1. Import the Iris data set from the sklearn.datasets module.
2. Explain, in your own words, what the load_iris() function returns.

In [3]:
# Save the iris dataset in the iris object
iris = sk.datasets.load_iris()

In [4]:
iris.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

The load_iris() function returns a dictionary-like object which contains multiple obkects, including: the iris dataset and the metadata of the dataset. This keys of this object are 'data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'.

# Task 2: Explore the Data Structure
Print and explain the shape of the data set, the first and last 5 rows of the data, the feature names, and the target classes.

In [5]:
# Display the shape of the Iris data
iris.data.shape

(150, 4)

The Iris dataset contains 150 rows (each sample is in a separate row) and 4 collums (each collumn corresponds to sepal length, sepal width, petal length and petal width).

In [6]:
# Display the first 5 rows
iris.data[0:5,]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

The dataset above contains info about sepal length, sepal width, petal length and petal width for the first 5 samples in the dataset.

In [7]:
# Display the last five rows
iris.data[-5:len(iris.data),]

array([[6.7, 3. , 5.2, 2.3],
       [6.3, 2.5, 5. , 1.9],
       [6.5, 3. , 5.2, 2. ],
       [6.2, 3.4, 5.4, 2.3],
       [5.9, 3. , 5.1, 1.8]])

The dataset above contains info about sepal length, sepal width, petal length and petal width for the last 5 samples in the dataset.

In [8]:
# Display the feature names
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

The feature_names object gives the explanation what does each collumn in the dataset represents.

In [9]:
# Display the sample types in the dataset
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

The target_names shows what sample types are present in the dataset.

# Task 3: Summarize the Data
For each feature in the dataset, calculate and display:  
**mean  
minimum  
maximum  
standard deviation  
median**  

In [53]:
# Transpose the iris dataset to separate the values of each feature in a separate list
iris_transposed = iris.data.transpose()

# Store the values of each feature in separate objects
sepal_length_list = iris_transposed[0]
sepal_width_list = iris_transposed[1]
petal_length_list = iris_transposed[2]
petal_width_list = iris_transposed[3]

In [67]:
# Calculate summary statistics of the dataset and store in the dictionary
def summary_statistics(dataset):

    # Create empty dictionary to store summary stats
    adict = {}

    # Compute summary stats and store in the dictionary
    adict["mean"] = round(dataset.mean(), 2)
    adict["standard_deviation"] = round(dataset.std(), 2)
    adict["minima"] = round(dataset.min(), 2)
    adict["maximum"] = round(dataset.max(), 2)
    adict["median"] = round(np.median(dataset), 2)

    return(adict)


In [68]:
#Compute summary stats of each feature
slength_sum_stats = summary_statistics(sepal_length_list)
swidth_sum_stats = summary_statistics(sepal_width_list)
plength_sum_stats = summary_statistics(petal_length_list)
pwidth_sum_stats = summary_statistics(petal_width_list)

In [71]:
# show_sum_stats prints summary statistics of a feature
# while using values stored in the dictionary
def show_sum_stats(feature_name, adict):
    
    print(f'''
Summary statistics of a {feature_name}
Mean: {adict["mean"]}
Standard deviation: {adict["standard_deviation"]}
Minima: {adict["minima"]}
Maxima: {adict["maximum"]}
Median: {adict["median"]}
        ''')

In [72]:
show_sum_stats("sepal length", slength_sum_stats)


Summary statistics of a sepal length
Mean: 5.84
Standard deviation: 0.83
Minima: 4.3
Maxima: 7.9
Median: 5.8
        


In [73]:
show_sum_stats("sepal width", swidth_sum_stats)


Summary statistics of a sepal width
Mean: 3.06
Standard deviation: 0.43
Minima: 2.0
Maxima: 4.4
Median: 3.0
        


In [74]:
show_sum_stats("petal length", plength_sum_stats)


Summary statistics of a petal length
Mean: 3.76
Standard deviation: 1.76
Minima: 1.0
Maxima: 6.9
Median: 4.35
        


In [75]:
show_sum_stats("petal width", pwidth_sum_stats)


Summary statistics of a petal width
Mean: 1.2
Standard deviation: 0.76
Minima: 0.1
Maxima: 2.5
Median: 1.3
        
